lib/core/sdam/topology.js

```diff
@@ -998,13 +999,13 @@ function resetServerState(server, error, options) {
       new ServerDescription(server.description.address, null, { error })
     );
   }
-
-  if (options.clearPool && server.pool) {
-    server.pool.reset(() => resetState());
+  if (options.clearPool && server.s.pool) {
+    server.s.pool.reset(() => resetState());
     return;
   }
 
   resetState();
+  server.monitor();
```

Comment:
I can't remember, why do we want to only kick off monitoring in this case? Should this be moved inside `resetState`?

Reply:
The spec says
```
When the client sees a "not master" or "node is recovering" error it MUST replace the server's description with a default ServerDescription of type Unknown. It MUST store useful information in the new ServerDescription's error field, including the error message from the server.

Multi-threaded and asynchronous clients MUST request an immediate check of the server. Unlike in the "network error" scenario above, a "not master" or "node is recovering" error means the server is available but the client is wrong about its type, thus an immediate re-check is likely to provide useful information.
```

Reply:
Right, but doesn't that mean we want to kick off the server check both when we reset the pool and not? I'm proposing moving `server.monitor()` up a few lines into the commonly called `resetState()` function